src/solvers/lowering/byte_operators.cpp

```diff
@@ -19,6 +19,168 @@ Author: Daniel Kroening, kroening@kroening.com
+    for(std::size_t i = 0; i < *num_elements; ++i)
+    {
+      if(subtype_bits.has_value())
+      {
+        const std::size_t subtype_bits_int =
+          numeric_cast_v<std::size_t>(*subtype_bits);
+        bitvector_typet type(bitvector_expr.type().id(), subtype_bits_int);
+        operands.push_back(bv_to_expr(
+          extractbits_exprt(
+            bitvector_expr,
+            (i + 1) * subtype_bits_int - 1,
+            i * subtype_bits_int,
+            type),
+          array_type.subtype(),
+          ns));
+      }
+      else
+      {
+        operands.push_back(
+          bv_to_expr(bitvector_expr, array_type.subtype(), ns));
+      }
+    }
+
+    return array_exprt(std::move(operands), array_type);
+  }
+  else if(target_type.id() == ID_vector)
+  {
+    const vector_typet &vector_type = to_vector_type(target_type);
+    const std::size_t num_elements =
+      numeric_cast_v<std::size_t>(vector_type.size());
+    auto subtype_bits = pointer_offset_bits(vector_type.subtype(), ns);
+
+    exprt::operandst operands;
+    operands.reserve(num_elements);
+    for(std::size_t i = 0; i < num_elements; ++i)
+    {
+      if(subtype_bits.has_value())
+      {
+        const std::size_t subtype_bits_int =
+          numeric_cast_v<std::size_t>(*subtype_bits);
+        bitvector_typet type(bitvector_expr.type().id(), subtype_bits_int);
+        operands.push_back(bv_to_expr(
+          extractbits_exprt(
+            bitvector_expr,
+            (i + 1) * subtype_bits_int - 1,
+            i * subtype_bits_int,
+            type),
+          vector_type.subtype(),
+          ns));
+      }
+      else
+      {
+        operands.push_back(
+          bv_to_expr(bitvector_expr, vector_type.subtype(), ns));
+      }
+    }
+
+    return vector_exprt(std::move(operands), vector_type);
+  }
+  else if(target_type.id() == ID_complex)
+  {
+    const complex_typet &complex_type = to_complex_type(target_type);
+    auto subtype_bits = pointer_offset_bits(complex_type.subtype(), ns);
+    std::size_t subtype_bits_int;
+    if(subtype_bits.has_value())
+      subtype_bits_int = numeric_cast_v<std::size_t>(*subtype_bits);
+    else
+      subtype_bits_int =
+        to_bitvector_type(bitvector_expr.type()).get_width() / 2;
+
+    bitvector_typet type(bitvector_expr.type().id(), subtype_bits_int);
+
+    return complex_exprt(
+      bv_to_expr(
+        extractbits_exprt(bitvector_expr, subtype_bits_int - 1, 0, type),
+        complex_type.subtype(),
+        ns),
+      bv_to_expr(
+        extractbits_exprt(
+          bitvector_expr, subtype_bits_int * 2 - 1, subtype_bits_int, type),
+        complex_type.subtype(),
+        ns),
+      complex_type);
+  }
+  else if(target_type.id() == ID_string)
+  {
+    // this works, but is somewhat questionable
+    return typecast_exprt(bitvector_expr, target_type);
+  }
+  else
+  {
+    PRECONDITION_WITH_DIAGNOSTICS(
+      false, "bv_to_expr does not yet support ", target_type.id_string());
```

Comment:
That's not a precondition, use `UNEXPECTEDCASE`

Reply:
There is `UNHANDLED_CASE`, but the documentation suggests that this should not be used, and it wouldn't provide diagnostics either. So I'm inclined to leave this as is.